### PR TITLE
Adding functional test cases for generating lower case URLs

### PR DESCRIPTION
**Pull request by kulmugdha**
@rynowak Please have a look.

### CODE REVIEW

File: Mvc.sln [hunk 1]
```diff
@@ -803,7 +806,6 @@ Global
 		{42C5D417-4060-48F4-BB28-E9E179007779}.Release|Mixed Platforms.Build.0 = Release|Any CPU
 		{42C5D417-4060-48F4-BB28-E9E179007779}.Release|x86.ActiveCfg = Release|Any CPU
 		{42C5D417-4060-48F4-BB28-E9E179007779}.Release|x86.Build.0 = Release|Any CPU
-<<<<<<< HEAD
```
Comment (rynowak) [inline Mvc.sln]: You have merge conflicts :laughing:

File: test/Microsoft.AspNet.Mvc.FunctionalTests/RoutingTests.cs [hunk 2]
```diff
@@ -1372,6 +1372,30 @@ public class RoutingTests
             Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
         }
 
+        [Theory]
+        [InlineData("LowercaseUrls_Blog/Generatelink", "/api/employee/getemployee/marykae?LastName=McDonald")]
+        [InlineData("LowercaseUrls_Blog/ShowPosts", "/lowercaseurls_blog/showposts")]
+        [InlineData("LowercaseUrls_Blog/edit/BlogPost", "/lowercaseurls_blog/edit/blogpost")]
+        [InlineData("Api/Employee/List", "/api/employee/list")]
+        [InlineData("Api/Employee/GetEmployee/JohnDoe", "/api/employee/getemployee/johndoe")]
+        public async Task GenerateLowerCaseUrlsTests(string path, string expectedUrl)
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/RoutingTests.cs]: Consider just creating a new file for this test. `RoutingLowerCaseUrlTest`

File: test/Microsoft.AspNet.Mvc.FunctionalTests/RoutingTests.cs [hunk 3]
```diff
@@ -1372,6 +1372,30 @@ public class RoutingTests
             Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
         }
 
+        [Theory]
+        [InlineData("LowercaseUrls_Blog/Generatelink", "/api/employee/getemployee/marykae?LastName=McDonald")]
+        [InlineData("LowercaseUrls_Blog/ShowPosts", "/lowercaseurls_blog/showposts")]
+        [InlineData("LowercaseUrls_Blog/edit/BlogPost", "/lowercaseurls_blog/edit/blogpost")]
+        [InlineData("Api/Employee/List", "/api/employee/list")]
+        [InlineData("Api/Employee/GetEmployee/JohnDoe", "/api/employee/getemployee/johndoe")]
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/RoutingTests.cs]: Please add some comments here about what these cover

File: test/Microsoft.AspNet.Mvc.FunctionalTests/project.json [hunk 4]
```diff
@@ -54,7 +54,8 @@
         "MvcTagHelpersWebSite": "1.0.0",
         "Microsoft.Framework.ConfigurationModel.Json": "1.0.0-*",
         "xunit.runner.kre": "1.0.0-*",
-        "Microsoft.AspNet.WebUtilities": "1.0.0-*"
+        "Microsoft.AspNet.WebUtilities": "1.0.0-*",
+        "LowercaseUrlsWebSite": "1.0.0-*"
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/project.json]: Keep these in alphabetical order (put your new one where it belongs in the order)

File: test/WebSites/LowercaseUrlsWebSite/Controllers/BlogController.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        }
+
+        public string Edit(string postName)
+        {
+            return Url.Action();
+        }
+
+        public string GenerateLink()
+        {
+            return Url.Action("GetEmployee", "LowercaseUrls_Employee", new { Name = "MaryKae" , LastName = "McDonald"});
```
Comment (rynowak) [inline test/WebSites/LowercaseUrlsWebSite/Controllers/BlogController.cs]: comment here to the effect of - Including extra data so it ends up in the query string

File: test/WebSites/LowercaseUrlsWebSite/Startup.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            var configuration = app.GetTestConfiguration();
+
+            app.UseServices(services =>
+            {
+                services.AddMvc(configuration);
+
+                services.Configure<RouteOptions>(routeOptions => routeOptions.LowercaseUrls = true);
+            });
+
+            app.UseErrorReporter();
```
Comment (rynowak) [inline test/WebSites/LowercaseUrlsWebSite/Startup.cs]: Please don't use this unless you have a specific need for it. it really degrades the troubleshooting experience when something goes wrong

**Comment by rynowak** [comment]
:shipit: after addressing minor comments

File: Mvc.sln (hunk 1 above)
Comment (kulmugdha) [inline Mvc.sln]: Yupp I need to fix it, but I thought I'll do it when I am ready to check in once and for all.

**Comment by rynowak** [comment]
@kulmugdha is there anything else to do here?

**Comment by kulmugdha** [comment]
No I have merged this change already, just forgot to close this one.

**Pull request closed without merging**